packages/kbn-test/src/functional_tests/tasks.js

```diff
@@ -95,6 +95,8 @@ export async function runTests(options) {
       try {
         es = await runElasticsearch({ config, options: opts });
         await runKibanaServer({ procs, config, options: opts });
+        // wait 5 seconds to let kibana startup completes
+        await silence(log, 5000);
```

Comment:
We wait for the log message `wait: /http server running/,` in `runKibanaServer()` before resolving the promise, please update that check if we need to wait longer. Adding a 5 second delay might work for now but there are a lot of things that can make startup take longer than 5 seconds and this will just be a source of flakiness

Reply:
> We wait for the log message wait: /http server running/, in runKibanaServer() before resolving the promise,

Yea, I saw that (being on this FTR problem for the past 5 hours) but that's the last message we can wait for, there isn't anything relevant (or, at all) after that. 

For some context, the issue is quite complicated, but basically with core going more sync, there are less returns to the event loop, and some background starts that were firing before the server is ready are no longer (most relevant one would be that the initial license check is not complete when the first FTR test kicks in), so we get 503 responses from Kibana in xpack FTR tests, as the security auth handler has no license to check against (because surprinsingly, in some test suites, the first http request from FTR to Kibana kicks in one millisecond after the server starts up, at least if the CI timestamps are correct). The roundtrip with ES to retrieve the license takes more than that (because fired later now that we 'block' the event loop loading every plugins synchronously)  

Instead of this hacky temporary 5sec wait, we should now perform requests to the `/status` endpoint and wait for Kibana status to be green before starting the FTR tests. But I wasn't confident doing so in the current PR, and was going to open an issue to do it as a follow-up, if that's alright with you? 

> Adding a 5 second delay might work for now but there are a lot of things that can make startup take longer than 5 seconds

The whole purpose of making server-side plugin loading sync is because nope, we don't want that 'things might take longer'. Waiting 300ms would even be plenty enough time for the CI env, so is 5sec should be way more than enough until we implement a proper status check.

Reply:
I mean, we have react renders that take over 5 seconds to complete and cause flakiness in the Jest Unit tests. There is often intense load on CI, so I really don't think that there's anything we could do to guarantee that a 5 second delay will be enough. I'm not totally opposed to a status check to replace the log message check, but if the Kibana server can't know when it's ready to serve requests and log a message stating so that seems like a bug in the design... There must be some way for the server to be able to report that it's ready.

Reply:
> There must be some way for the server to be able to report that it's ready.

Yea, that would be the `/status` endpoint.

`http server running` means the http server is ready. The plugins have not necessarily performed some initial background tasks and may not be green yet.

One of these tasks is the retrieval of the license from ES, performed by the `licensing` plugin. The fact is, the implementation of the security's `auth` http handler does not await this initial license, and make the assumption that at the time the server's ready, the license has been fetched. With FTR running the tests instantly after greping the `http server running`, some test suites performing http calls to authenticated endpoint in before hooks get a 503 error coming from the auth hook.

I did a change to explicitly await the license here: https://github.com/elastic/kibana/pull/89562/commits/fdf73feb0ae7358cf6f2286848f4c77b81909ecf. This (reverted) commit was also fixing the problem, but the changes are more impactful, for a temporary solution, than a one line await delay(5) than does not affect production code.

However, the correct way to do this awaiting should probably be to use the `/status` endpoint of the server. That's meant to be its primary purpose.

But, this status check would need to be done either here: https://github.com/elastic/kibana/blob/320dd5f9bf50a6d09e8fce46dc9177e50275c8a6/packages/kbn-test/src/functional_tests/tasks.js#L96-L100, or in the FTR runner. We need to also implement proper status reporting for the `licensing` plugin, as the plugin does not currently await for its initial license check to return a green status.

So, I was going to create a follow-up to discuss more in details our options and how exactly we want to address that, and, in the meantime, I felt like adding the delay to the FTR run script was the more pragmatic solution.

But if you think that's not acceptable, we can address that directly in this PR

cc @legrego @restrry

Reply:
I'm alright with a temporary solution, the only issue I have is that failures from flakiness caused at this level are likely to be spread out and hard to trace back to the cause, so ideally we won't have a temporary solution in place for long. What if the status endpoint logged the first time all plugins went green or something? Could we rely on that as a signal rather than trying to figure out what the right URL for the Kibana server is here?

Reply:
>  The roundtrip with ES to retrieve the license takes more than that (because fired later now that we 'block' the event loop loading every plugins synchronously)

it is worth noting that even in the old implementation we were not protected from the fact that the FTR request will come before we received a license from the server. 

> rather than trying to figure out what the right URL for the Kibana server is here?

We have access to the config to read protocol, port, certificate authorities and other setting to query Kibana. Don't we? 
 https://github.com/elastic/kibana/blob/170a2956c84d46262a3ae310f89aee6f2580860a/packages/kbn-test/src/functional_tests/lib/run_elasticsearch.js#L24

> What if the status endpoint logged the first time all plugins went green or something?

@elastic/kibana-operations owns `@kbn/test` package, so I see no reason not to do it if you prefer this option. But let's do it in the follow-up PR.

Reply:
I created https://github.com/elastic/kibana/issues/89828 to discuss/confirm which approach we exactly want to take.